format/nexus.py

```diff
@@ -535,25 +535,39 @@ class BeamFactory(object):
     A class to create a beam model from NXmx stuff
     """
 
-    def __init__(self, obj, index=None):
+    def __init__(self, obj):
```

Comment:
Since you're changing the class constructor anyway, can we get rid of the class entirely? It is a pseudo class with a (trivial) constructor that then only offers a single function. Can we turn it into a proper function? Cache references are possible using attributes on the function itself or cache decorators.

Reply:
Hi again I think refactoring is out of scope of this request.  I have no opinion on classes vs. functions for trivial cases.